mcs/class/Mono.Posix/Mono.Unix.Native/Syscall.cs

```diff
@@ -1416,6 +1440,177 @@ public struct Linger {
 		public int l_linger;
 	}
 
+	[Map]
+	[StructLayout (LayoutKind.Sequential)]
+	[CLSCompliant (false)]
+	public struct InAddr : IEquatable<InAddr> {
+		public uint s_addr;
+
+		public InAddr (byte b0, byte b1, byte b2, byte b3)
+		{
+			s_addr = 0;
+			this[0] = b0;
```

Comment:
> This seems inefficient, as you'll be evaluating the ArgumentOufOfRangeException check 4 times, when you know you shouldn't need any.

Ok, I'll make sure the constructor doesn't do the range check.
 
> I also wonder -- and I'm not sure how to verify :-( -- what byte ordering constraints there are, if any. I believe that you intend for s_addr to be in Network Byte Order...but should it be? I don't know what it should be, but this does give me pause.

Yes, `s_addr` is supposed to be network byte order. `Mono_Posix_FromInAddr()`/`Mono_Posix_ToInAddr()` assume that the layout of `struct InAddr` in C# and `struct in_addr` in C is the same.

I think that it should be network byte order, as most people think of an IP address not as a 32-bit number but as 4 8-bit numbers. One possibility would also be to make s_addr private (then the fact that it is stored in network byte order is no longer visible to the caller) and make callers rely on the setters/getters or the `Copy*()` methods.